api/src/main/java/org/pmiops/workbench/cohorts/CohortMaterializationService.java

```diff
@@ -170,6 +170,12 @@ public MaterializeCohortResponse materializeCohort(@Nullable CohortReview cohort
             String.format("Use of pagination token %s with new parameter values", paginationToken));
       }
     }
+    // Grab the next pagination token here, because searchRequest can be mutated during query
+    // execution, which will make the hash of SearchRequest no longer match; see
+    // [RW-844].
+    // TODO: consider pagination based on cursor / values rather than offset
+    PaginationToken newToken = PaginationToken.of(offset + pageSize, paginationParameters);
```

Comment:
nit: Right now this depends on the utility eagerly hashing the input parameters, else the bug wouldn't be fixed. I had to click through to verify that's what was happening.

I'd probably do the `toBase64()` here instead to remove this implementation dependency.